custom_components/ai_thermostat/climate.py

```diff
@@ -275,7 +278,19 @@ def _async_startup(*_):
 					self._target_temp = float(old_state.attributes[ATTR_TEMPERATURE])
 			if not self._hvac_mode and old_state.state:
 				self._hvac_mode = old_state.state
-		
+			if old_state.attributes.get(ATTR_STATE_LAST_CHANGE) is not None:
+				self.last_change = old_state.attributes.get(ATTR_STATE_LAST_CHANGE)
+			if old_state.attributes.get(ATTR_STATE_WINDOW_OPEN) is not None:
+				self.window_open = old_state.attributes.get(ATTR_STATE_WINDOW_OPEN)
+			if old_state.attributes.get(ATTR_STATE_DAY_TEMP) is not None:
+				self.daytime_temp = old_state.attributes.get(ATTR_STATE_DAY_TEMP)
+			if old_state.attributes.get(ATTR_STATE_SUMMER) is not None:
+				self.is_summer = old_state.attributes.get(ATTR_STATE_SUMMER)
+			if old_state.attributes.get(ATTR_STATE_NIGHT_MODE) is not None:
+				self.night_status = old_state.attributes.get(ATTR_STATE_NIGHT_MODE)
+				if self.night_status:
+					self._target_temp = float(self.night_temp)				
```

Comment:
It's not safe to cast this unprotected.

Do a 
```
import numbers
if self.night_temp and isinstance(self.night_temp, numbers.Number):
    self._target_temp = float(self.night_temp)
else:
    #fail somehow
```

We shouldn't try to parse strings here. :)

Reply:
Think its more usable for the user, can be rewritten if the night mode gets replaced

Reply:
Our users should not see this code, except your cast fails :D

self.night_temp might be None at this point in time. Casting is _not safe_.

Please fix your code.